pkgs/_macro_host/lib/src/package_config.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:dart_model/dart_model.dart';
+import 'package:package_config/package_config.dart' as package_config;
+
+/// Reads a package config to determine information about macros.
+class PackageConfig {
+  final Uri uri;
+
+  PackageConfig({required this.uri});
+
+  /// Checks whether [name] is a macro annotation.
+  ///
+  /// If so, returns the qualified name of the macro implementation.
+  ///
+  /// If not, returns `null`.
+  ///
+  /// This is a placeholder implementation until `language/3728` is
+  /// implemented. It expects macros to be marked by a comment in the
+  /// annotation package `pubspec.yaml` that looks like this:
+  ///
+  /// ```
+  /// # macro <annotation name> <implementation name>
+  /// ```
+  ///
+  /// For example:
+  ///
+  /// ```
+  /// # macro declare_x_macro.dart#DeclareX package:_test_macros/declare_x_macro.dart#DeclareXImplementation
+  /// ```
+  QualifiedName? lookupMacroImplementation(QualifiedName name) {
+    var packageName = name.uri;
+    if (packageName.startsWith('package:') && packageName.contains('/')) {
+      packageName = packageName.substring('package:'.length);
+      packageName = packageName.substring(0, packageName.indexOf('/'));
+    } else {
+      throw ArgumentError('Name must start "package:" and have a path: $name');
+    }
+    final packageConfig = package_config.PackageConfig.parseBytes(
+        File.fromUri(uri).readAsBytesSync(), uri);
+    final libraryQualifiedName =
```

Comment:
nit: This is really a path not just a name. It also doesn't include the `lib/` which it should, to allow macros to be defined outside of `lib/` (and below, I would expect the full path not the lib relative path in the `# macro` comments.

Reply:
Done.